esp_isotp/inc/esp_isotp.h

```diff
@@ -30,14 +39,44 @@
  *     esp_isotp_send(handle, data, size);
  *
  *     // Check for received data (non-blocking)
- *     uint16_t received_size;
+ *     uint32_t received_size;
  *     if (esp_isotp_receive(handle, buffer, sizeof(buffer), &received_size) == ESP_OK) {
  *         // Complete message received
  *     }
  *
  *     vTaskDelay(pdMS_TO_TICKS(5));
  * }
  * ```
+ *
+ * ## Event-Driven Usage (Callback Mode)
+ * ```c
+ * void my_rx_callback(const uint8_t *data, uint32_t size, void *user_arg) {
+ *     // Process received data - copy immediately if needed beyond callback
+ *     printf("Received %d bytes\n", size);
+ *     // Handle received data...
+ * }
+ *
+ * void my_tx_callback(uint32_t tx_size, void *user_arg) {
+ *     printf("Transmitted %d bytes successfully\n", tx_size);
+ * }
+ *
+ * esp_isotp_config_t config = {
+ *     .tx_id = 0x123, .rx_id = 0x124,
+ *     .tx_buffer_size = 4095, .rx_buffer_size = 4095,
+ *     .use_extended_id = false,
+ *     .rx_callback = my_rx_callback,  // Event-driven mode
+ *     .tx_callback = my_tx_callback,
+ *     .callback_arg = &my_context
+ * };
+ *
+ * esp_isotp_handle_t handle;
+ * esp_isotp_new_transport(twai_node, &config, &handle);
+ *
+ * while (1) {
+ *     esp_isotp_poll(handle);  // MUST still call regularly!
+ *     vTaskDelay(pdMS_TO_TICKS(5));
+ * }
+ * ```
```

Comment:
I wonder if our doc build system will handle this part nicely. If not, I would suggest to put the example code in the programming guide.